src/LayerPlan.cpp

```diff
@@ -2152,6 +2153,238 @@ void LayerPlan::addLinesInGivenOrder(
     }
 }
 
+LayerPlan::TravelDurations LayerPlan::computeTravelDurations(const GCodeExport& gcode, const ExtruderTrain& extruder, const GCodePath& path, const coord_t z_hop_height) const
+{
+    TravelDurations travel_durations;
+
+    if (z_hop_height > 0)
+    {
+        const Velocity z_hop_speed = extruder.settings_.get<Velocity>("speed_z_hop");
+        travel_durations.z_hop = (z_hop_height / z_hop_speed) / 1000.0;
+    }
+
+    if (! path.points.empty())
+    {
+        const Velocity& travel_speed = path.config.getSpeed();
+        Point2LL start_position = gcode.getPosition().toPoint2LL();
+        for (const Point3LL& travel_point : path.points)
+        {
+            const Point2LL travel_point_2d = travel_point.toPoint2LL();
+            const coord_t travel_segment_length = vSize(travel_point_2d - start_position);
+            travel_durations.travel += (travel_segment_length / travel_speed) / 1000.0;
+            start_position = travel_point_2d;
+        }
+    }
```

Comment:
Untested and a suggestion, it should be a bit more performant avoids some extra calculations
```suggestion
    if (! path.points.empty())
    {
        const Velocity travel_speed = path.config.getSpeed();
        const double factor = 1.0 / travel_speed / 1000.0;
        Point2LL start = gcode.getPosition().toPoint2LL();
        auto total_distance = std::accumulate(
            path.points.begin(), path.points.end(),
            0.0,
            [&start](double acc, const Point3LL& pt)
            {
                const Point2LL current = pt.toPoint2LL();
                double seg = vSize(current - start);
                start = current;
                return acc + seg;
            });
        travel_durations.travel += total_distance * factor;
    }
```